### ISSUE TITLE
Newline at end of text line in contents property is not included when appended to wiki file

### ISSUE THREAD

**Issue by msorokowski** [rfennell/AzurePipelines#988]
**Describe the bug**
I'm using the WIKIUpdater to append content to an existing wiki page. The content is specified in the **contents** property (instead of reading from a file).

The newline at the end of a text line in the **contents** property is not included when it's appended to the wiki page content.

Thus, a newline is not included on the last line of the appended content in the wiki page. So when this task is executed a couple of times, it will end up appending the content to the end of the last line instead of the next line since the last line will not have a newline .

Here's a screenshot of the YAML (displaying CTRL characters to show the CR LFs). This example has a newline at the end of the desired text line to append, plus a newline on the next line (for good measure, but is expected to be excluded since the literal style with clip should maintain a single newline at the end of the string).

![image](https://user-images.githubusercontent.com/13974987/112674650-c7196400-8e2b-11eb-8a19-a7253a84f8ac.png)

However, the newline is not added to the wiki page content.

Here's a screenshot of the wiki page content (and preview) after executing the task 3 times:

![image](https://user-images.githubusercontent.com/13974987/112672497-2cb82100-8e29-11eb-89e4-31a487937d33.png)

**Expected behavior**
Expecting the newline at the end of the content line to be copied to the wiki page content.

**Where is the issue seen:**
 - Azure DevOps Services (VSTS)

**Comment by rfennell** [comment]
I will have to look into that, but  wonder if it is 'feature' of how multi-line  parameters are parsed

**Comment by msorokowski** [comment]
I did another test... If there's 2 lines containing text, the newline at the end of the first line is copied. The trailing spaces and newlines after the last line of text are clipped.

![image](https://user-images.githubusercontent.com/13974987/112694641-4fa5fd80-8e48-11eb-915d-d7d1a9a391ca.png)

So "Line 2" does end up on another line. But the next time it's executed, "Line 1" is appended to the end of the previous "Line 2".

Here's the raw markdown and preview of the results after 2 executions:

![image](https://user-images.githubusercontent.com/13974987/112694808-9dbb0100-8e48-11eb-877a-0263f98d442c.png)

**Comment by rfennell** [comment]
I think that supports my assumption there is some trimming of the text in a multiline YAML. I think the best option is to add a new parameter to the task to optionally insert a new line between appended/prepended files.

**Comment by msorokowski** [comment]
I guess there’s a few use cases out there that might warrant prepending a newline before the content, and/or appending a newline after the content.

Something for you to think about, but I think one or the other would resolve the issue I’ve encountered.

**Comment by rfennell** [comment]
The updated task is being released as 1.24.3. There is now an extra parameter `insertLinefeed`, if set to `true` an extra linefeed is added between the old and new content when appending or pretending content.

```
- task: WikiUpdaterTask@1
  inputs:
    ... other params
    repo: 'dev.azure.com/richardfennell/Git%20project/_git/Git-project.wiki'
    replaceFile: false
    appendToFile: true
    insertLinefeed: true
    dataIsFile: false
    contents: |
      - $(build.buildnumber)
      - Line 1
      - Line 2
```

**Comment by msorokowski** [comment]
Thanks! I just tested the new feature and all is good.
